app/lib/account/default_auth_provider.dart

```diff
@@ -320,10 +397,31 @@ abstract class BaseAuthProvider extends AuthProvider {
       _logger.warning('JWT rejected, email_verified = "$emailVerified"');
       return null; // missing email (probably missing 'email' scope)
     }
+    final nonce = r['nonce'];
+    if (nonce != null && nonce is! String) {
+      _logger.warning('JWT rejected, bad `nonce`');
+      return null;
+    }
+    if (expectedNonce != null && nonce != expectedNonce) {
```

Comment:
This should use fixed-time comparison. Otherwise it might be vulnerable to a timing attack.

It'd probably be hard to do in practice, since you probably can't create enough Google signed JWTs to launch an actual timing attack :D

Reply:
Updated.